app/controllers/menus_controller.rb

```diff
@@ -1,74 +1,129 @@
 class MenusController < ApplicationController
+  before_filter :authenticate_user!
+  before_action :set_hotel
+  before_action :set_order_id, only: [:view_menu, :add_order_item, :reduce_order_item ]
   before_action :set_menu, only: [:show, :edit, :update, :destroy]
 
-  # GET /menus
-  # GET /menus.json
   def index
-    @menus = Menu.all
-  end
-
-  # GET /menus/1
-  # GET /menus/1.json
-  def show
-  end
-
-  # GET /menus/new
-  def new
+    @menus = @hotel.menus.all
     @menu = Menu.new
   end
-
-  # GET /menus/1/edit
-  def edit
+  def new
+    @menu = @hotel.menus.build
   end
-
-  # POST /menus
-  # POST /menus.json
   def create
-    @menu = Menu.new(menu_params)
-
-    respond_to do |format|
-      if @menu.save
-        format.html { redirect_to @menu, notice: 'Menu was successfully created.' }
-        format.json { render :show, status: :created, location: @menu }
-      else
-        format.html { render :new }
-        format.json { render json: @menu.errors, status: :unprocessable_entity }
-      end
+    @menu = @hotel.menus.build(menu_params)
+    @menu.hotel_id=@hotel.id
+    if @menu.save
+    @menus = @hotel.menus.all
     end
   end
 
-  # PATCH/PUT /menus/1
-  # PATCH/PUT /menus/1.json
   def update
-    respond_to do |format|
-      if @menu.update(menu_params)
-        format.html { redirect_to @menu, notice: 'Menu was successfully updated.' }
-        format.json { render :show, status: :ok, location: @menu }
-      else
-        format.html { render :edit }
-        format.json { render json: @menu.errors, status: :unprocessable_entity }
-      end
-    end
+   if  @menu.update(menu_params)
+    @menus = @hotel.menus.all
+   end
   end
 
-  # DELETE /menus/1
-  # DELETE /menus/1.json
   def destroy
     @menu.destroy
-    respond_to do |format|
-      format.html { redirect_to menus_url, notice: 'Menu was successfully destroyed.' }
-      format.json { head :no_content }
+  end
+  def view_menu
+
+    @menus = @hotel.menus.all
+    @order_items=@hotel.orders.find(session[:order_id]).try(:order_items) || []
+       
+    if session[:order_id]!=nil
+    @order=Order.find(session[:order_id])
+    if @order.hotel!=@hotel       
+      @order.order_items.all.destroy
+    end    
```

Comment:
@shaunak-amura  when user click on 'view menu' button it redirects to menu_item selection page.  when user click on add menu_item button that time new order object is created which is stored in session.when user trying add another menu/_items that time it checks whether order_id is in session or not? if yes then i will add menu_item with order reference which is in session else i will create new order object and that will b stored in session

suppose after menu item selection user goes back change the hotel, and click on view menu then while loading order item list it will display order items for specific hotel and rest items will be deleted.